tests/ha/control_pod/test_control_pod_softfailure.py

```diff
@@ -0,0 +1,273 @@
+        self.del_users = self.mgnt_ops.create_account_users(nusers=self.num_users)
+        LOGGER.info("Precondition: Get header token befor soft-failure")
+        self.header = self.csm_obj.get_headers(self.csm_user, self.csm_passwd)
+        LOGGER.info("Done: Setup operations.")
+
+    def teardown_method(self):
+        """
+        This function will be invoked after each test function in the module.
+        """
+        LOGGER.info("STARTED: Teardown Operations.")
+        if self.s3_clean:
+            LOGGER.info("Cleanup: Cleaning created s3 accounts and buckets.")
+            resp = self.ha_obj.delete_s3_acc_buckets_objects(self.s3_clean)
+            assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Revert back to default single control pod per cluster if more replicas are "
+                    "created.")
+        resp = self.node_master_list[0].create_pod_replicas(num_replica=self.def_replica,
+                                                            deploy=self.deply_name)
+        assert_utils.assert_true(resp[0], resp[1])
+        if os.path.exists(self.test_dir_path):
+            system_utils.remove_dirs(self.test_dir_path)
+        LOGGER.info("Done: Teardown completed.")
+
+    # pylint: disable=too-many-statements
+    # pylint: disable-msg=too-many-locals
+    @pytest.mark.ha
+    @pytest.mark.lc
+    @pytest.mark.tags("TEST-45498")
+    def test_45498(self):
+        """
+        Verify IAM user creation/deletion and IOs on already created IAM user while performing
+        control pod soft-failure in loop one by one for N control pods.
+        """
+        LOGGER.info("STARTED: Verify IAM user creation/deletion & IOs on already created IAM user "
+                    "while performing control pod soft-failure in loop 1 by 1 for N control pods.")
+
+        LOGGER.info("Step 1: Perform IOs before soft-failure.")
+        self.test_prefix = 'test-45498'
+        resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(self.s3_clean.values())[0],
+                                                    log_prefix=self.test_prefix, skipcleanup=True,
+                                                    nsamples=2, nclients=2)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 1: Performed IOs before soft-failure.")
+
+        LOGGER.info("Scale replicas for control pod to %s", self.repl_num)
+        resp = self.node_master_list[0].create_pod_replicas(num_replica=self.repl_num,
+                                                            deploy=self.deply_name)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Total number of control pods in the cluster are: %s", self.repl_num)
+
+        iam_output = Queue()
+        io_output = Queue()
+        event = threading.Event()
+
+        LOGGER.info("Step 2: Starting independent background threads for IOs on already created "
```

Comment:
are we sure we need IOs in case of control pod? cc: @apurwamohite

Reply:
We can do only bucket CRUDs, but she has already added IOs, so we can keep them in this case only.